assigner/tests/config_test.py

```diff
@@ -0,0 +1,92 @@
+from assigner.tests.utils import AssignerTestCase
+
+from assigner.config.versions import validate, get_version, upgrade, ValidationError
+from assigner.config.upgrades import UPGRADES
+from assigner.config.schemas import SCHEMAS
+
+
+CONFIGS = [
+    { # Version 0
+        "token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 1
+        "gitlab-token": "xxx gitlab token xxx",
+        "gitlab-host": "https://git.gitlab.com",
+        "namespace": "assigner-testing",
+        "semester": "2016-SP",
+        "roster": [],
+    },
+    { # Version 2
+        "version": 2,
+        "backend": {
+            "name": "gitlab",
+            "token": "xxx gitlab token xxx",
+            "host": "https://git.gitlab.com",
+        },
```

Comment:
@brhoades What do you think of this configuration schema? If you like it, I'll update the rest of the code to work with it.

Reply:
The schema will go something like this: the `backend` field is a JSON object with at least one key, `name`. The remainder of the data will be something that we can pass to the `Backend` class and it will be able to pull what it needs out of that.

So, for the `mock` backend, the config would look like
```
"backend" : { "name" : "mock" }
```

Seem sensible?

Reply:
That seems fine to me. No more YAML 😭

Reply:
I want to make sure I understand you right. Are you referring to the schema here, where we double check the functions are properly enforcing schema version differences? Or do you mean in general :D?

Reply:
I mean for the newest schema, in preparation for #102  😄

Reply:
This looks good. Will allow nonbreaking backend additions.

Reply:
Cool -- in that case I'll propagate the change through `BaseRepo` et. al.